go/vt/discovery/healthcheck.go

```diff
@@ -166,6 +170,8 @@ type tabletAliasString string
 
 // HealthCheck declares what the TabletGateway needs from the HealthCheck
 type HealthCheck interface {
+	TabletRecorder
+
```

Comment:
Why are we embedding an interface into another? Do we have implementations of TabletRecorder that don't implement HealthCheck?

Reply:
> Do we have implementations of TabletRecorder that don't implement HealthCheck?

No, we don't.

> Why are we embedding an interface into another?

This was done so we can use the healthcheck (`discovery.HealthCheck`) as a tablet recorder in `tx throttler`: https://github.com/vitessio/vitess/blob/4654a771c2868fd0a6eab8ac96af39f82e5c532c/go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go#L199

Ultimately, the topo watcher in the `discovery` package uses a `TabletRecorder`.

Previously, the `LegacyHealthCheck` interface was embedding `LegacyTabletRecorder`:
https://github.com/vitessio/vitess/blob/d037f24c8c4efd52cf70fac2e11fcb2941ce0a32/go/vt/discovery/legacy_healthcheck.go#L285-L289

Reply:
we should not have to embed it for things to work.